design/src/rfcs/rfc0021_dependency_versions.md

```diff
@@ -0,0 +1,71 @@
+RFC: Dependency Versions
+========================
+
+> Status: RFC
+>
+> Applies to: Client and Server
+
+This RFC outlines how Rust dependency versions are selected for the smithy-rs project, and
+strives to meet the following semi-conflicting goals:
+
+- Dependencies are secure
+- Vended libraries have dependency ranges that overlap other Rust libraries as much as possible
+
+When in conflict, the security goal takes priority over the compatibility goal.
+
+Categorization of Crates
+------------------------
+
+The Rust crates within smithy-rs can be divided up into two categories:
+
+1. **Library Crates:** Crates that are published to crates.io with the intention that other projects
+   will depend on them via their `Cargo.toml` files. This category does NOT include binaries that
+   are published to crates.io with the intention of being installed with `cargo install`.
```

Comment:
We don't have any of these do we?

Reply:
Right now, there's just the `cargo-check-external-types` tool, but I'm hoping to break that out into a separate GItHub repository sometime soon.